libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -90,16 +91,14 @@ public static ParsedMediaType parseMediaType(String headerValue) {
                     if (paramsAsString.isEmpty()) {
                         continue;
                     }
-                    // intentionally allowing to have spaces around `=`
-                    // https://tools.ietf.org/html/rfc7231#section-3.1.1.1 disallows this
-                    String[] keyValueParam = elements[i].trim().split("=");
-                    if (keyValueParam.length == 2) {
-                        String parameterName = keyValueParam[0].toLowerCase(Locale.ROOT).trim();
-                        String parameterValue = keyValueParam[1].toLowerCase(Locale.ROOT).trim();
-                        parameters.put(parameterName, parameterValue);
-                    } else {
+                    //spaces are allowed between parameters, but not between '=' sign
+                    String[] keyValueParam = paramsAsString.split("=");
+                    if (keyValueParam.length != 2 || hasSpaces(keyValueParam[0]) || hasSpaces(keyValueParam[1])) {
```

Comment:
I think for `keyValueParam[0]` you only want to check trailing whitespace, and for `keyValueParam[1]` you only want to check for leading whitespace. `trim()` checks for both.

Reply:
I think you can use https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isWhitespace(char) for the first and last characters to check them independently.

Reply:
good idea, I was meant to refactor with trim as it was unnecessarily creating garbage